lib/ansible/modules/cloud/amazon/elb_target_group.py

```diff
@@ -477,8 +479,8 @@ def create_or_update_target_group(connection, module):
                 module.fail_json_aws(e, msg="Couldn't update target group")
 
         # Do we need to modify targets?
-        if module.params.get("modify_targets"):
-            if module.params.get("targets"):
+        if module.params.get("purge_targets"):
+            if module.params.get("targets") == []:
```

Comment:
I think `if module.params.get("targets"):` is right here. It indicates that the module needs to compare the current targets with the targets the user provided and remove any that the user hasn't provided. If the comparison within this `if` is wrong and the wrong targets get removed then that should be fixed.

If I provide
```
...
purge_targets: true
targets:
      - Id: i-01234567
        Port: 80
      - Id: i-98765432
        Port: 80
```
and then in a second playbook:
```
...
purge_targets: true
targets:
      - Id: i-01234567
        Port: 80
```
I would expect target i-98765432 to be purged after the second playbook. But with this patch this if would now be false, so the path on line 538 would be used and all targets would be purged regardless of the target I want to keep.

I was thinking this line should be changed: https://github.com/ansible/ansible/pull/39715/files#diff-c008972f0ccccd4a4d9b8c1de95c4d08L538 to `elif module.params.get("targets") == []` or `elif module.params.get("targets") is not None`